src/api/records.test.tsx

```diff
@@ -0,0 +1,228 @@
+    const xRow = x[i];
+    const yRow = y[i];
+
+    const xKeys = Object.keys(xRow);
+    const yKeys = Object.keys(yRow);
+
+    for (let i = 0; i < xKeys.length; i++) {
+      if (xKeys[i] !== yKeys[i]) return false;
+    }
+
+    if (xRow.timestamp !== yRow.timestamp) return false;
+    if (xRow.shotnum !== yRow.shotnum) return false;
+    if (xRow.activeArea !== yRow.activeArea) return false;
+    if (xRow.activeExperiment !== yRow.activeExperiment) return false;
+  }
+
+  return true;
+};
+
+describe('records api functions', () => {
+  let mockData: Record[];
+  let params: URLSearchParams;
+  let state: PreloadedState<RootState>;
+
+  beforeEach(() => {
+    mockData = testRecords;
+    params = new URLSearchParams();
+    state = getInitialState();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('useRecordsPaginated', () => {
+    beforeEach(() => {
+      (axios.get as jest.Mock).mockResolvedValue({
+        data: mockData,
+      });
+    });
+
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('sends axios request to fetch records and returns successful response', async () => {
+      params.append('limit', '25');
+      params.append('skip', '0');
+
+      const { result } = renderHook(() => useRecordsPaginated(), {
+        wrapper: hooksWrapperWithProviders(),
+      });
+
+      await waitFor(() => {
+        expect(result.current.isSuccess).toBeTruthy();
+      });
+
+      expect(axios.get).toHaveBeenCalledWith(
+        '/records',
+        expect.objectContaining({ params })
+      );
+      expect(
+        dataResponsesEqual(result.current.data, testRecordRows)
+      ).toBeTruthy();
+    });
+
+    it('can send sort and date range parameters as part of request', async () => {
+      state = {
+        ...getInitialState(),
+        table: { ...getInitialState().table, sort: { timestamp: 'asc' } },
+        search: {
+          ...getInitialState().search,
+          dateRange: {
+            fromDate: '2022-01-01 00:00:00',
+            toDate: '2022-01-02: 00:00:00',
+          },
+        },
+      };
+
+      params.append('limit', '25');
+      params.append('skip', '0');
+      params.append('order', 'timestamp asc');
+      params.append(
+        'conditions',
+        "{$and:[{'metadata.timestamp':'$gt':'2022-01-01 00:00:00','$lt':'2022-01-02 00:00:00'}]"
+      );
+
+      const { result } = renderHook(() => useRecordsPaginated(), {
+        wrapper: hooksWrapperWithProviders(state),
+      });
+
+      await waitFor(() => {
+        expect(result.current.isSuccess).toBeTruthy();
+      });
+
+      expect(axios.get).toHaveBeenCalledWith(
+        '/records',
+        expect.objectContaining({ params })
+      );
+      expect(
+        dataResponsesEqual(result.current.data, testRecordRows)
+      ).toBeTruthy();
+    });
+
+    it.skip('sends axios request to fetch records and throws an appropriate error on failure', async () => {
```

Comment:
Why is this test skipped?

Reply:
Truth be told, I didn't think it was worth it atm due to errors simply console logging "Got error". I thought to leave it until we have a proper error handling system, similar to `handleICATError()`

Reply:
Can we maybe use `todo` then instead of `skip`? `skip` makes me think the test isn't working

Reply:
Yep, can do